From 586893408fa1c274844941c74757c3364c0f701e Mon Sep 17 00:00:00 2001
From: Boris Ranto <branto@redhat.com>
Date: Wed, 4 Mar 2015 15:23:53 +0100
Subject: [PATCH 1/2] Switch google/* includes to gperftools/* includes

---
 src/perfglue/heap_profiler.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/perfglue/heap_profiler.cc b/src/perfglue/heap_profiler.cc
index 6b079b8..cdd5ccb 100644
--- a/src/perfglue/heap_profiler.cc
+++ b/src/perfglue/heap_profiler.cc
@@ -12,8 +12,8 @@
  * 
  */
 
-#include <google/heap-profiler.h>
-#include <google/malloc_extension.h>
+#include <gperftools/heap-profiler.h>
+#include <gperftools/malloc_extension.h>
 #include "heap_profiler.h"
 #include "common/environment.h"
 #include "common/LogClient.h"
-- 
1.9.3

