From 69e888fa5c103588ee9f23ac4cec4bfc2285eb08 Mon Sep 17 00:00:00 2001
From: Boris Ranto <branto@redhat.com>
Date: Fri, 13 Jan 2017 01:15:42 +0100
Subject: [PATCH] hack: do not test for libxfs, assume it is present

---
 configure.ac | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 32e273f..050e754 100644
--- a/configure.ac
+++ b/configure.ac
@@ -882,7 +882,7 @@ AC_ARG_WITH([libxfs],
   [AS_HELP_STRING([--without-libxfs], [disable libxfs use by FileStore])],
   [],
   [with_libxfs=yes])
-AS_IF([test "x$with_libxfs" != "xno"], [
+AS_IF([test "x$with_libxfs" = "hack..."], [
   # xfs/xfs.h presence and XFS_XFLAG_EXTSIZE define
   AC_CHECK_HEADER([xfs/xfs.h], [], AC_MSG_ERROR(
     [xfs/xfs.h not found (--without-libxfs to disable)]))
@@ -900,6 +900,7 @@ AS_IF([test "x$with_libxfs" != "xno"], [
     AC_MSG_ERROR([XFS_XFLAG_EXTSIZE not found (--without-libxfs to disable)])
   ])
 ])
+AC_DEFINE([HAVE_LIBXFS], [1], [Define to 1 if you have libxfs])
 AM_CONDITIONAL(WITH_LIBXFS, [test "x$with_libxfs" != "xno"])
 
 # use libzfs
-- 
2.7.4

