From b204da4e28a0238547c99ee1bf4080ae4d62ad33 Mon Sep 17 00:00:00 2001
From: Boris Ranto <branto@redhat.com>
Date: Wed, 8 Feb 2017 22:38:12 +0100
Subject: [PATCH] librbd/Journal: include WorkQueue since we use it

Signed-off-by: Boris Ranto <branto@redhat.com>
---
 src/librbd/Journal.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/librbd/Journal.h b/src/librbd/Journal.h
index 4b70b99..2490975 100644
--- a/src/librbd/Journal.h
+++ b/src/librbd/Journal.h
@@ -11,6 +11,7 @@
 #include "common/Cond.h"
 #include "common/Mutex.h"
 #include "common/Cond.h"
+#include "common/WorkQueue.h"
 #include "journal/Future.h"
 #include "journal/JournalMetadataListener.h"
 #include "journal/ReplayEntry.h"
-- 
2.9.3

