diff -up ceph-0.67.3/src/common/crc32c-intel.c.secondary ceph-0.67.3/src/common/crc32c-intel.c
--- ceph-0.67.3/src/common/crc32c-intel.c.secondary	2013-10-07 10:15:22.000000000 +0200
+++ ceph-0.67.3/src/common/crc32c-intel.c	2013-10-07 10:16:00.000000000 +0200
@@ -7,8 +7,8 @@
 #include <sys/wait.h>
 
 
-/* this probably isn't specific enough for x86_64?  fix me someday */
-#ifdef __LP64__
+/* this probably is specific enough for x86_64 */
+#ifdef __x86_64__
 
 /*
  *  * Based on a posting to lkml by Austin Zhang <austin.zhang@intel.com>
@@ -98,7 +98,7 @@ int ceph_have_crc32c_intel(void)
 	return 0;
 }
 
-#else /* __LP64__ */
+#else /* __x86_64__ */
 
 uint32_t ceph_crc32c_le_intel(uint32_t crc, unsigned char const *data, unsigned length)
 {
