diff -up ceph-0.26/src/rgw/rgw_cache.h.josef ceph-0.26/src/rgw/rgw_cache.h
--- ceph-0.26/src/rgw/rgw_cache.h.josef	2011-04-05 15:19:29.364573666 -0700
+++ ceph-0.26/src/rgw/rgw_cache.h	2011-04-05 15:20:11.946014058 -0700
@@ -98,12 +98,12 @@ int RGWCache<T>::obj_stat(std::string& b
   bufferlist bl;
   uint64_t size;
   time_t mtime;
+  int64_t t;
 
   int r = cache.get(name, bl);
   if (r == 0) {
     bufferlist::iterator iter = bl.begin();
     ::decode(size, iter);
-    int64_t t;
     ::decode(t, iter);
     mtime = (time_t)t;
     goto done;
@@ -113,7 +113,7 @@ int RGWCache<T>::obj_stat(std::string& b
     return r;
   bl.clear();
   ::encode(size, bl);
-  int64_t t = (int64_t)mtime;
+  t = (int64_t)mtime;
   ::encode(t, bl);
   cache.put(name, bl);
 done:
