diff -urp ceph-0.87/src/include/encoding.h ceph-0.87-pm/src/include/encoding.h
--- ceph-0.87/src/include/encoding.h	2014-10-29 19:09:14.000000000 +0100
+++ ceph-0.87-pm/src/include/encoding.h	2015-01-29 02:47:10.392860344 +0100
@@ -286,7 +286,7 @@ inline void decode(T &o, bufferlist& bl)
 #include <deque>
 #include <vector>
 #include <string>
-#include <boost/optional.hpp>
+#include <boost/optional/optional_io.hpp>
 
 #ifndef _BACKWARD_BACKWARD_WARNING_H
 #define _BACKWARD_BACKWARD_WARNING_H   // make gcc 4.3 shut up about hash_*
diff -urp ceph-0.87/src/os/DBObjectMap.h ceph-0.87-pm/src/os/DBObjectMap.h
--- ceph-0.87/src/os/DBObjectMap.h	2014-10-29 19:09:14.000000000 +0100
+++ ceph-0.87-pm/src/os/DBObjectMap.h	2015-01-29 02:46:59.915044699 +0100
@@ -17,7 +17,7 @@
 #include "common/Mutex.h"
 #include "common/Cond.h"
 #include "common/simple_cache.hpp"
-#include <boost/optional.hpp>
+#include <boost/optional/optional_io.hpp>
 
 /**
  * DBObjectMap: Implements ObjectMap in terms of KeyValueDB
diff -urp ceph-0.87/src/osd/ECBackend.cc ceph-0.87-pm/src/osd/ECBackend.cc
--- ceph-0.87/src/osd/ECBackend.cc	2014-10-29 19:09:19.000000000 +0100
+++ ceph-0.87-pm/src/osd/ECBackend.cc	2015-01-29 02:56:47.662703397 +0100
@@ -13,7 +13,7 @@
  */
 
 #include <boost/variant.hpp>
-#include <boost/optional.hpp>
+#include <boost/optional/optional_io.hpp>
 #include <iostream>
 #include <sstream>
 
@@ -81,7 +81,7 @@ ostream &operator<<(ostream &lhs, const
   lhs << "read_result_t(r=" << rhs.r
       << ", errors=" << rhs.errors;
   if (rhs.attrs) {
-    lhs << ", attrs=" << rhs.attrs;
+    lhs << ", attrs=" << (bool)rhs.attrs;
   } else {
     lhs << ", noattrs";
   }
diff -urp ceph-0.87/src/osd/ECBackend.h ceph-0.87-pm/src/osd/ECBackend.h
--- ceph-0.87/src/osd/ECBackend.h	2014-10-29 19:09:14.000000000 +0100
+++ ceph-0.87-pm/src/osd/ECBackend.h	2015-01-29 02:47:01.716013012 +0100
@@ -18,7 +18,7 @@
 #include "OSD.h"
 #include "PGBackend.h"
 #include "osd_types.h"
-#include <boost/optional.hpp>
+#include <boost/optional/optional_io.hpp>
 #include "erasure-code/ErasureCodeInterface.h"
 #include "ECTransaction.h"
 #include "ECMsgTypes.h"
diff -urp ceph-0.87/src/osd/ECTransaction.cc ceph-0.87-pm/src/osd/ECTransaction.cc
--- ceph-0.87/src/osd/ECTransaction.cc	2014-10-29 19:09:19.000000000 +0100
+++ ceph-0.87-pm/src/osd/ECTransaction.cc	2015-01-29 02:47:00.229039175 +0100
@@ -13,7 +13,7 @@
  */
 
 #include <boost/variant.hpp>
-#include <boost/optional.hpp>
+#include <boost/optional/optional_io.hpp>
 #include <iostream>
 #include <vector>
 #include <sstream>
diff -urp ceph-0.87/src/osd/ECTransaction.h ceph-0.87-pm/src/osd/ECTransaction.h
--- ceph-0.87/src/osd/ECTransaction.h	2014-10-29 19:09:14.000000000 +0100
+++ ceph-0.87-pm/src/osd/ECTransaction.h	2015-01-29 02:47:01.383018871 +0100
@@ -19,7 +19,7 @@
 #include "PGBackend.h"
 #include "osd_types.h"
 #include "ECUtil.h"
-#include <boost/optional.hpp>
+#include <boost/optional/optional_io.hpp>
 #include "erasure-code/ErasureCodeInterface.h"
 
 class ECTransaction : public PGBackend::PGTransaction {
diff -urp ceph-0.87/src/osd/osd_types.h ceph-0.87-pm/src/osd/osd_types.h
--- ceph-0.87/src/osd/osd_types.h	2014-10-29 19:09:14.000000000 +0100
+++ ceph-0.87-pm/src/osd/osd_types.h	2015-01-29 02:47:01.400018572 +0100
@@ -22,7 +22,7 @@
 #include <stdio.h>
 #include <memory>
 #include <boost/scoped_ptr.hpp>
-#include <boost/optional.hpp>
+#include <boost/optional/optional_io.hpp>
 
 #include "include/rados/rados_types.hpp"
 
diff -urp ceph-0.87/src/osd/ReplicatedPG.h ceph-0.87-pm/src/osd/ReplicatedPG.h
--- ceph-0.87/src/osd/ReplicatedPG.h	2014-10-29 19:09:14.000000000 +0100
+++ ceph-0.87-pm/src/osd/ReplicatedPG.h	2015-01-29 02:47:05.062954123 +0100
@@ -17,7 +17,7 @@
 #ifndef CEPH_REPLICATEDPG_H
 #define CEPH_REPLICATEDPG_H
 
-#include <boost/optional.hpp>
+#include <boost/optional/optional_io.hpp>
 #include <boost/tuple/tuple.hpp>
 
 #include "include/assert.h" 
diff -urp ceph-0.87/src/common/map_cacher.hpp ceph-0.87-pm/src/common/map_cacher.hpp
--- ceph-0.87/src/common/map_cacher.hpp	2014-10-29 19:09:14.000000000 +0100
+++ ceph-0.87-pm/src/common/map_cacher.hpp	2015-01-29 02:48:12.007776242 +0100
@@ -16,7 +16,7 @@
 #define MAPCACHER_H
 
 #include <boost/scoped_ptr.hpp>
-#include <boost/optional.hpp>
+#include <boost/optional/optional_io.hpp>
 #include "include/memory.h"
 #include <set>
 #include <map>
